### PR TITLE
Change `as_factor()` to convert ordered factors to plain factors

### LINKED ISSUE #216: Feature Request: handle ordered factors within as_factor()
Currently, there is not a way to convert an `"ordered" "factor"` into a plain `"factor"` within `{forcats}`.

Hadley requested that PR https://github.com/tidyverse/forcats/pull/215#issuecomment-542714204 be converted into an issue. For easy reading, here's the comment:

> Based on our current understanding of S3, I think it's clear that these functions should be as.factor() and as.ordered(). as.ordered() works currently, but as.factor() does not. We can't fix as.factor(), but we can fix as_factor(). Doing that 100% correctly is a bit fiddly, so would you mind filing an issue?

### PR DESCRIPTION
**Pull request by robinsones**
Closes #216. 

I used `structure()` as that's what `as_factor.character` uses. Added two tests and a NEWS item as well.

### CODE REVIEW

File: R/as_factor.R [hunk 1]
```diff
@@ -34,7 +34,12 @@ as_factor <- function(x, ...) {
 #' @rdname as_factor
 #' @export
 as_factor.factor <- function(x, ...) {
-  x
+  structure(
+    x,
+    label = attr(x, "label", exact = TRUE),
```
Comment (hadley) [inline R/as_factor.R]: Can you please order these alphabetically?

File: tests/testthat/test-as_factor.R [hunk 2]
```diff
@@ -16,3 +16,10 @@ test_that("supports NA (#89)", {
   x <- c("a", "z", "g", NA)
   expect_equal(as_factor(x), fct_inorder(x))
 })
+
+test_that("converts ordered factor into plain factor (#216)", {
+  x <- factor(c("a", "z", "g"), ordered = TRUE)
+  x <- as_factor(x)
+  expect_false(is.ordered(x))
+  expect_equal(attributes(x)$class, "factor")
```
Comment (hadley) [inline tests/testthat/test-as_factor.R]: I think I'd just do `expect_s3_class(as_factor(x), "factor", exact = "TRUE")` (i.e. you only need one test)

File: R/as_factor.R (hunk 1 above)
Comment (robinsones) [reply R/as_factor.R]: done

File: tests/testthat/test-as_factor.R (hunk 2 above)
Comment (robinsones) [reply tests/testthat/test-as_factor.R]: done

**Comment by hadley** [comment]
Thanks @robinsones!

**Comment by hadley** [comment]
Unfortunately I had to revert this PR because it causes haven to fail R CMD check. That's totally a bug with haven, but because haven has C warnings that are slightly outside of my control, it's unfortunately easier to revert here than to squeeze out a quick haven release.

**Pull request merged**